src/lib/y2partitioner/immediate_unmount.rb

```diff
@@ -0,0 +1,115 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast/i18n"
+require "yast2/popup"
+require "y2partitioner/device_graphs"
+
+module Y2Partitioner
+  # Mixin
+  module ImmediateUnmount
+  private
+
+    extend Yast::I18n
+
+    # Sets textdomain
+    def included(_target)
+      textdomain "storage"
+    end
+
+    def immediate_unmount(device, full_message: nil, note: nil, allow_continue: true)
+      loop do
+        case confirm_unmount(device, full_message, note, allow_continue)
+        when :unmount
+          unmount(device) || next
+        when :continue
+          break true
+        when :cancel
+          break false
+        end
+      end
+    end
+
+    def confirm_unmount(device, full_message, note, allow_continue)
+      headline = Yast::Label.WarningMsg
+
+      message = immediate_unmount_message(device, full_message, note, allow_continue)
+
+      buttons = inmmediate_unmount_buttons(allow_continue)
+
+      Yast2::Popup.show(message, headline: headline, buttons: buttons, focus: :cancel)
+    end
+
+    def unmount(device)
+      device.mount_point.immediate_deactivate
+      true
+    rescue Storage::Exception
+      show_unmount_error
+      false
+    end
+
+    def show_unmount_error
+      message = _("The file system could not be unmounted")
```

Comment:
this is quite useless without any details why it could not be unmounted. It can be temporary issue, critical issue or whatever in between.

Reply:
I fear that we don't have the reason when libstorage-ng cannot unmount, but I will check.

Reply:
So exception does not contain anything useful?

Reply:
BTW, the old partitioner simply showed nothing in that case :)

Reply:
I have checked and it has the error reason. I have added it.